src/api_manager/service_control/proto.cc

```diff
@@ -50,6 +51,22 @@ namespace google {
 namespace api_manager {
 namespace service_control {
 
+namespace {
+
+// Maximum 36 byte string for UUID
+const int kMaxUUIDBufSize = 40;
+
+// Generates a UUID string
+std::string GenerateUUID() {
```

Comment:
we may not need this

the operation_id for the by_consumer one can be:  request:uuid + 1